services/intersection-api/api/src/test/java/us/dot/its/jpo/ode/api/controllers/intersections/IntersectionControllerTest.java

```diff
@@ -0,0 +1,131 @@
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import us.dot.its.jpo.ode.api.accessors.map.ProcessedMapRepository;
+import us.dot.its.jpo.ode.api.models.IntersectionReferenceData;
+import us.dot.its.jpo.ode.api.services.PermissionService;
+import us.dot.its.jpo.ode.api.services.PostgresService;
+
+public class IntersectionControllerTest {
+
+    private IntersectionController controller;
+
+    @Mock
+    ProcessedMapRepository processedMapRepo;
+
+    @Mock
+    PostgresService postgresService;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        controller = new IntersectionController(
+                processedMapRepo,
+                postgresService);
+    }
+
+    @Test
+    void testGetIntersectionsTestData() {
+        ResponseEntity<List<IntersectionReferenceData>> response = controller.getIntersections(null, true);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody()).hasSize(1);
+        assertThat(response.getBody().get(0).getIntersectionID()).isEqualTo(12109);
+    }
+
+    @Test
+    void testGetIntersectionsByOrganization() {
+        IntersectionReferenceData intersection = new IntersectionReferenceData();
+        intersection.setIntersectionID(1);
+        List<IntersectionReferenceData> allIntersections = Collections.singletonList(intersection);
+
+        when(processedMapRepo.getIntersectionIDs()).thenReturn(allIntersections);
+        when(postgresService.getAllowedIntersectionIdsByOrganization("org"))
+                .thenReturn(Collections.singletonList(1));
+
+        ResponseEntity<List<IntersectionReferenceData>> response = controller.getIntersections("org", false);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody()).contains(intersection);
+    }
+
+    @Test
+    void testGetIntersectionsByEmail() {
+        IntersectionReferenceData intersection = new IntersectionReferenceData();
+        intersection.setIntersectionID(2);
+        List<IntersectionReferenceData> allIntersections = Arrays.asList(intersection);
+
+        when(processedMapRepo.getIntersectionIDs()).thenReturn(allIntersections);
+
+        String username = "testuser";
+        try (MockedStatic<PermissionService> mockedStatic = Mockito.mockStatic(PermissionService.class)) {
+            mockedStatic.when(() -> PermissionService.getUsername(any())).thenReturn(username);
+            when(postgresService.getAllowedIntersectionIdsByEmail(username))
+                    .thenReturn(Collections.singletonList(2));
+
+            ResponseEntity<List<IntersectionReferenceData>> response = controller.getIntersections(null, false);
+            assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+            assertThat(response.getBody()).contains(intersection);
```

Comment:
suggestion: we're configuring a single intersection to return for the given email, it would make sense to ensure our return body has a single object which _is_ the intersection, rather than contains the intersection (and possibly other data).